lib/debug-pages.js

```diff
@@ -0,0 +1,164 @@
+
+const debuggerSharedState = getDebuggerSharedState()
+const configuredHandlers = getConfiguredHandlersSharedState()
+
+const jsonReplacer = (key, value) => {
+  if (typeof value === 'bigint') {
+    return Number(value)
+  }
+  return value
+}
+
+function setupHandlerForFile (filePath, mimeType) {
+  return () => {
+    return {
+      statusCode: 200,
+      headers: {
+        'Content-Type': mimeType
+      },
+      body: fs.createReadStream(filePath)
+    }
+  }
+}
+
+const globalConstants = { runAmockVersion }
+
+function replaceVariables (source, ...variableSources) {
+  const allSourcesMixedTogether = Object.assign({}, ...variableSources)
+  return Object.keys(allSourcesMixedTogether).reduce((accumulator, key) => {
+    const find = `{{${key}}}`
+    const rawValue = allSourcesMixedTogether[key]
+    const processedValue = typeof rawValue === 'object' ? `<code><pre>${JSON.stringify(rawValue, jsonReplacer, 2)}</pre></code>` : rawValue
+    return accumulator.replaceAll(find, processedValue)
+  }, source)
+}
+
+function filterConfiguredHandlers (handlers, filterFn) {
+  const output = {}
+  Object.keys(handlers).forEach(methodOrDefault => {
+    if (methodOrDefault === '__default__') {
+      if (filterFn(handlers[methodOrDefault])) {
+        output.__default__ = handlers[methodOrDefault]
+      }
+      return
+    }
+    Object.keys(handlers[methodOrDefault]).forEach(key => {
+      const handlersArray = handlers[methodOrDefault][key]
+      if (!Array.isArray(handlersArray)) {
+        console.error('Expected array, got', handlersArray)
+        console.error(`Lookup was [${key}] in`, handlers[methodOrDefault])
+      }
+      handlersArray.filter(filterFn).forEach(handlerConfig => {
+        output[methodOrDefault] = output[methodOrDefault] || {}
+        output[methodOrDefault][key] = output[methodOrDefault][key] || []
+        output[methodOrDefault][key].push(handlerConfig)
+      })
+    })
+  })
+  return output
+}
+
+function getSnapshotListHtml () {
+  const defaultLi = '<li><a href="/__debugger__">Current state (not a snapshot)</a></li>'
+  return [defaultLi].concat(Object.keys(debuggerSharedState.snapshots)
+    .map(key => `<li><a href="/__debugger__?snapshot=${encodeURIComponent(key)}">${key}</a></li>`))
+    .join('\n')
+}
+
+export function clearSnapshotsEndpoint () {
+  debuggerSharedState.snapshots = []
+  const url = '/__debugger__#snapshots'
+  return {
+    statusCode: 302,
+    headers: {
+      Location: url
+    },
+    body: `Redirecting to ${url}`
+  }
+}
+
+function parseSnapshotJson (json) {
+  return JSON.parse(json)
+}
+
+export function debuggerSnapshotEndpoint (req) {
+  const snapshotName = [
+    new Date().toISOString(),
+    req.body?.name
+  ]
+    .filter(x => !!x)
+    .join(': ')
+
+  debuggerSharedState.snapshots[snapshotName] = {
+    latestMockRequest: JSON.stringify(debuggerSharedState.latestMockRequest || 'Not called the snapshot was made', jsonReplacer),
```

Comment:
Not sure the line 'Not called the snapshot was made' reads correctly? Maybe needs a comma?